tests/virtualization/xen/hotplugging.pm

```diff
@@ -22,37 +22,71 @@ use version_utils;
 sub run_test {
     my ($self) = @_;
     my $hypervisor = get_var('HYPERVISOR') // '127.0.0.1';
+    my ($sles_running_version, $sles_running_sp) = get_sles_release;
+
+    my %mac = ();
+    record_info "Virtual network", "Adding virtual network interface";
+    foreach my $guest (keys %xen::guests) {
+        $mac{$guest} = '00:16:3f:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
+        unless ($guest =~ m/hvm/i && is_sle('<=12-SP2') && check_var("XEN", "1")) {
+            my $persistent_config_option = '';
+            my $interface_model_option   = '';
+            if (get_var('VIRT_AUTOTEST')) {
+                if (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen')) {
+                    record_soft_failure 'bsc#1168124 Bridge network interface hotplugging has to be performed at the beginning.';
+                    $self->{test_results}->{$guest}->{"bsc#1168124 Bridge network interface hotplugging has to be performed at the beginning"}->{status} = 'SOFTFAILED';
+                    $persistent_config_option = '--persistent' if ($sles_running_version eq '11' && $sles_running_sp eq '4');
+                }
+                if (check_var('SYSTEM_ROLE', 'kvm') || check_var('HOST_HYPERVISOR', 'kvm')) {
+                    my $interface_model_option = '--model virtio';
+                    script_run "ssh root\@$guest modprobe acpiphp", 60 if ($guest =~ /^sles-11-sp4.*$/img);
+                    record_soft_failure 'bsc#1167828 Bridge network interface br0 hotplugging does not work with this ' . $guest if ($guest =~ /^sles-11-sp4.*$/img);
+                    $self->{test_results}->{$guest}->{"bsc#1167828 Bridge network interface br0 hotplugging does not work with this $guest"}->{status} = 'SOFTFAILED' if ($guest =~ /^sles-11-sp4.*$/img);
+                }
+            }
+            script_retry "ssh root\@$guest ip l | grep " . $xen::guests{$guest}->{macaddress}, delay => 60, retry => 3, timeout => 60;
+            assert_script_run "virsh attach-interface --domain $guest --type bridge ${interface_model_option} --source br0 --mac " . $mac{$guest} . " --live " . ${persistent_config_option};
```

Comment:
```suggestion
if (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen')) {
    assert_script_run "virsh attach-interface --domain $guest --type bridge  --source br0 --mac " . $mac{$guest} . " --live --persistent"};
} else {
    assert_script_run "virsh attach-interface --domain $guest --type bridge --model virtio --source br0 --mac " . $mac{$guest} . " --live"};
}
```

Reply:
I do think this change is necessary. Variable ${persistent_config_option} already implements this.

Reply:
This change will make the code cleaner. You should of course remove also `$interface_model_option` and `$persistent_config_option` as those variables are just making the code less readable.

Reply:
You can not remove all of them, and there will must be some other kind of duplicates weigh in after making the change, for example "assert_script_run" and "script_run" duplicates. Please think about it.

Reply:
You can remove them and the result is cleaner:
```perl
if (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen')) {
    assert_script_run "virsh attach-interface --domain $guest --type bridge  --source br0 --mac " . $mac{$guest} . " --live --persistent"};
} else {
    assert_script_run "virsh attach-interface --domain $guest --type bridge --model virtio --source br0 --mac " . $mac{$guest} . " --live"};
}
```

Reply:
If just for this specific simple case only, it does looks nicer to some extent. But for "--persistent", it is not always there for all xen scenarios(please refer back to original code). And, at the same time, it belongs to xen part together with other stuff including softfailure treatments for our use. So the better way is to combine them together and use variables to differentiate the real operations from each other. If you want to remove all dependent variables, the definite result is lots of if/else blocks and many variance of assert_script_run. I can not call them clean. The almost same applies to interface_model_option.

Reply:
Why don't we use ` --persistent` all the times on XEN and why don't we use `--model virtio` on KVM?

Reply:
"--persistent" only needed on specific xen host because some old version can only perform hotplugging with both live and persisent there, which apparently an error on newer version. "--model virtio" is default option to kvm and needs to be specified otherwise there will be error reporting. Please do not remove 9 lines and add 10 lines back, there is not too much meaning.

Reply:
OK. This work for all of us, let's keep it.